exporters/tests/test_pelorus_base_plugin.py

```diff
@@ -0,0 +1,342 @@
+#!/usr/bin/env python3
+#
+# Copyright Red Hat
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+
+import http
+import json
+import time
+from typing import Any, Awaitable
+from unittest.mock import AsyncMock, Mock, patch
+
+import pytest
+from pydantic import ValidationError, parse_obj_as
+from typing_extensions import override
+
+from webhook.models.pelorus_webhook import (
+    CommitTimePelorusPayload,
+    PelorusMetric,
+    PelorusMetricSpec,
+)
+from webhook.plugins.pelorus_handler_base import (
+    Headers,
+    HTTPException,
+    PelorusWebhookPlugin,
+    PelorusWebhookResponse,
+    Request,
+)
+
+
+@pytest.mark.parametrize(
+    "test_no,http_response,http_response_code",
+    [
+        (1, "Invalid Payload.", http.HTTPStatus.UNPROCESSABLE_ENTITY),
+        (2, "Invalid Payload.", "422"),
+        (3, None, http.HTTPStatus.UNPROCESSABLE_ENTITY),
+        (4, "Invalid Payload.", 99),
+        (5, "Invalid Payload.", 601),
+        (6, "Invalid Payload.", None),
+    ],
+)
+def test_pelorus_webhook_response(test_no, http_response, http_response_code):
+    """
+    Test for the PelorusWebhookResponse class. The response code must be a valid
+    HTTPStatus code. The http_response must be valid string.
+    """
+
+    # Test for proper message and response code is used.
+    if test_no < 3:
```

Comment:
break test in 2 to remove the `if` statement (test_pelorus_webhook_response_success and test_pelorus_webhook_response_error, for example)

Reply:
changed.